routers/web/repo/pull.go

```diff
@@ -1002,6 +1014,11 @@ func MergePullRequest(ctx *context.Context) {
 		ctx.ServerError("Merge", err)
 		return
 	}
+	// commit dbCtx
+	if err := committer.Commit(); err != nil {
+		ctx.ServerError("Commit", err)
+		return
+	}
```

Comment:
```suggestion
	// commit dbCtx
	if err := committer.Commit(); err != nil {
		ctx.ServerError("Commit", err)
		return
	}
	committer = nil
	dbCtx = nil
```

The logic is too complex here, it's better to reset these variables to prevent from misuse.